### DISCUSSION TITLE
classes/class_translation

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#178] [User-contributed notes]
# classes/class_translation

Inherits: Resource< RefCounted< Object Inherited By: OptimizedTranslation A language translation that maps a collection of strings to their individual translations. Description: Translation s are r...

https://docs.godotengine.org/en/stable/classes/class_translation.html

**Comment by FoxyFox909** [comment]
I wrote a little [function](https://gist.github.com/FoxyFox909/edd4fa8861416bea7a2f49040188e3c6) that will coalesce a bunch of CSV files into one single locale, PER language. That way, you can have CSV-based localization but have multiple CSV files rather than one monolithic one.

Note: Translations need to NOT be compressed, to work around [#38862](https://github.com/godotengine/godot/issues/38862)

**Comment by turbohermit** [comment]
Using get_plural_message or add_plural_message on this class will result in the following error: "Translation class doesn't handle plural messages. Calling add_plural_message() on a Translation instance is probably a mistake. 
  WARNING: Use a derived Translation class that handles plurals, such as TranslationPO class."

If that's by design, this documentation should reflect the intention and use of these functions. This documentation specifically states that it's useful for pluralization, but the base class seems to not actually support that functionality.

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/108617
